pkg/controller/servicebinding/binder.go

```diff
@@ -433,14 +449,28 @@ func (b *binder) removeContainer(container interface{}) (map[string]interface{},
 	return runtime.DefaultUnstructuredConverter.ToUnstructured(c)
 }
 
-// appendVolumeMounts append the binding volume in the template level.
-func (b *binder) appendVolumeMounts(volumeMounts []corev1.VolumeMount) []corev1.VolumeMount {
+func (b *binder) mountPath() string {
 	name := b.sbr.GetName()
-	mountPath := b.sbr.Spec.MountPathPrefix
+	mountPath := b.bindingRoot
+	fixedMountPath := false
 	if mountPath == "" {
-		mountPath = "/var/data"
+		mountPath = b.sbr.Spec.MountPath
+		if mountPath == "" {
+			mountPath = "/bindings"
+		} else {
+			fixedMountPath = true
+		}
 	}
 
+	if !fixedMountPath {
+		mountPath = path.Join(mountPath, name)
+	}
+	return mountPath
+}
+
+// appendVolumeMounts append the binding volume in the template level.
+func (b *binder) appendVolumeMounts(volumeMounts []corev1.VolumeMount, mountPath string) []corev1.VolumeMount {
```

Comment:
This logic is not handling changing the `mountPath` after the first reconciliation. Write a test to demonstrate this functionality.

Reply:
Done